packages/sources/src/components/tree-view/tree-view.demo.ts

```diff
@@ -80,4 +111,55 @@ export namespace TreeViewDemo {
       throw new Error('Updating items collection encountered an error.');
     });
   }
+
+  /**
+   * Updates all items in a tree view item collection.
+   * @param collection {TreeViewItem[]} the collection
+   * @param updater the updater function
+   */
+  function updateTree(
+    collection: TreeViewItem[],
+    updater: (item: TreeViewItem) => TreeViewItem
+  ): TreeViewItem[] {
+
+    return collection.map(root => {
+      const item = updater(root);
+      return {
+        ...item,
+        items: item.items ? updateTree(item.items, updater) : item.items
+      };
+    });
+  }
+
+  /**
+   * Opens all items in a tree view item collection that has a descendant with selected state.
+   * @param collection {TreeViewItem[]} the collection
+   * @param collapse {boolean} if true items with no selected descendants will be closed
+   */
+  function openItems(
+    collection: TreeViewItem[],
+    collapse: boolean
+  ): number {
+    const flatten = (data: any, level: number) => data ? data.flatMap((item: any) => ([{ level, item}, ...flatten(item.items, level + 1)])) : [];
+
+    const list: Array<{ level: number, item: TreeViewItem }> = flatten(collection, 0).reverse();
+
+    let levelOpen: boolean[] = [];
+
+    list.forEach((row: { level: number, item: TreeViewItem }) => {
+      if (levelOpen[row.level + 1]) {
+        row.item.open = true;
+        levelOpen[row.level + 1] = false;
+      }
+      else if (collapse) {
+        row.item.open = false;
+      }
+
+      if (row.item.selected || row.item.open) {
+        levelOpen[row.level] = true;
+      }
+    });
+
+    return list.filter((row: any) => row.item.selected).length;
```

Comment:
Ook hier denk ik dat de typering weg kan.

Reply:
Klopt, dit is helemaal niet nodig.